vllm/compilation/partition_rules.py

```diff
@@ -0,0 +1,102 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+
+from __future__ import annotations
+
+import contextlib
+
+import torch
+from torch._ops import OpOverload, OpOverloadPacket
+
+from vllm.logger import init_logger
+
+logger = init_logger(__name__)
+
+
+def _parse_operator_name(op_name: str) -> tuple[str, str, str]:
+    if not op_name:
+        raise ValueError("Operator name must be non-empty")
+
+    parts = op_name.split(".")
+    if len(parts) < 2:
+        raise ValueError(
+            f"Operator name '{op_name}' must include a namespace and operator"
+        )
+
+    namespace, remainder = parts[0], ".".join(parts[1:])
+    if not remainder:
+        raise ValueError(
+            f"Operator name '{op_name}' must include an operator identifier"
+        )
+
+    if "." in remainder:
+        operator, overload = remainder.split(".", 1)
+    else:
+        operator, overload = remainder, "default"
+    overload = overload or "default"
+    return namespace, operator, overload
+
+
+def _resolve_operator_overload(op_name: str):
+    namespace, operator, overload = _parse_operator_name(op_name)
+    target_overload = overload or "default"
+
+    try:
+        namespace_obj = getattr(torch.ops, namespace)
+        operator_obj = getattr(namespace_obj, operator)
+    except AttributeError as exc:
+        if not hasattr(torch.ops, namespace):
+            raise ValueError(f"Unknown operator namespace '{namespace}'") from exc
+        raise ValueError(f"Unknown operator '{namespace}::{operator}'") from exc
```

Comment:
Let's just use this instead of doing our own parsing: https://github.com/pytorch/pytorch/blob/c1f40d33c89b361a1edad17aa25cfff1ab4014fd/torch/_library/utils.py#L61

Reply:
This is much nicer!